instrumentation/kafka/kafka-clients/README.md

```diff
@@ -1,6 +1,145 @@
+* `OTEL_METRICS_EXPORTER`: the list of exporters to be used for metrics. If you don't use metrics then it has to be set to `none`;
+
+Instead of using the above environment variables, it is also possible to use corresponding system properties to be set programmatically or on the command line.
+They are `otel.service.name`, `otel.traces.exporter` and `otel.metrics.exporter`.
+
+#### SDK builders: programmatic configuration
+
+In order to build your own `OpenTelemetry` instance and not relying on autoconfiguration, it is possible to do so by using the SDK builders programmatically.
+The OpenTelemetry SDK dependency is needed in order to have such SDK builders classes available.
+
+```xml
+<dependency>
+    <groupId>io.opentelemetry</groupId>
+    <artifactId>opentelemetry-sdk</artifactId>
+    <version>OPENTELEMETRY_RELEASE</version>
+</dependency>
+```
+
+In order to get an `OpenTelemetry` instance and registers it globally so that it can be used by the Kafka clients, follow the official documentation [here](https://opentelemetry.io/docs/instrumentation/java/manual/#set-up) for setting up the SDK.
+
+### Using interceptors
+
+The Kafka clients API provides a way to "intercept" messages before they are sent to the brokers as well as messages received from the broker before being passed to the application.
+The OpenTelemetry instrumented Kafka library provides two interceptors to be configured to add tracing information automatically.
+The interceptor class has to be set in the properties bag used to create the Kafka client.
+
+Use the `TracingProducerInterceptor` for the producer in order to create a "send" span automatically, each time a message is sent.
+
+```java
+props.setProperty(ProducerConfig.INTERCEPTOR_CLASSES_CONFIG, TracingProducerInterceptor.class.getName());
+```
+
+Use the `TracingConsumerInterceptor` for the consumer in order to create a "receive" span automatically, each time a message is received.
+
+```java
+props.setProperty(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, TracingConsumerInterceptor.class.getName());
+```
+
+### Wrapping clients
+
+The other way is by wrapping the Kafka client with a tracing enabled Kafka client.
+
+Assuming you have a `Producer<K, V> producer` instance, you can wrap it in the following way.
+
+```java
+KafkaTelemetry telemetry = KafkaTelemetry.create(GlobalOpenTelemetry.get());
+Producer<String, String> tracingProducer = telemetry.wrap(producer);
+```
+
+Then use the `tracingProducer` as usual for sending messages to the Kafka cluster.
+
+Assuming you have a `Consumer<K, V> consumer` instance, you can wrap it in the following way.
+
+```java
+KafkaTelemetry telemetry = KafkaTelemetry.create(GlobalOpenTelemetry.get());
+Consumer<String, String> tracingConsumer = telemetry.wrap(this.consumer);
+```
+
+Then use the `tracingConsumer` as usual for receiving messages from the Kafka cluster.
+
+## Using agent
```

Comment:
The agent setup is also described on the [opentelemetry.io](https://opentelemetry.io/docs/instrumentation/java/automatic/#setup) page, we can omit it here.

Reply:
In this case I would not omit the entire section as for the same reason; user having content here to start quickly.
I would leave the initial introduction, adding to run producer and consumer as described in the "link to the opentelemetry agent doc" you are referring to.

Reply:
I think this section is fine to stay, but there's really no need to write docs on the agent itself that are unrelated to this library. I'd just shorten the words by saying "make sure you have the java agent", link to the autoinstrumentation docs that talk about it, and then have the examples of how to call the agent with the kafka instrumentation.